### PR TITLE
Default the pkcs11 code to use sha256 for OAEP padding

### PR DESCRIPTION
**Pull request by stefanberger**
This PR modifies the pkcs11 code to default to sha256

### CODE REVIEW

File: crypto/pkcs11/pkcs11helpers.go [hunk 1]
```diff
@@ -431,15 +426,18 @@ func EncryptMultiple(pubKeys []interface{}, data []byte) ([]byte, error) {
...
 //     } ,
 //     {
 //        "version": 0,
 //        "blob": <base64 encoded RSA OAEP encrypted blob>,
-//        "hash": <hash used for OAEP other than 'sha256'>
+//        "hash": <hash used for OAEP other than 'sha1'>
 //     } ,
 //     [...]
 // }
+// Note: More recent versions of this code explicityly write 'sha1'
```
Comment (lumjjb) [inline crypto/pkcs11/pkcs11helpers.go:438]: nit : explicitly

File: crypto/pkcs11/pkcs11helpers_test.go [hunk 2]
```diff
@@ -133,6 +133,8 @@ module:
 	p11pubkeyfileobj.Uri.SetModuleDirectories(p11conf.ModuleDirectories)
 	p11pubkeyfileobj.Uri.SetAllowedModulePaths(p11conf.ModuleDirectories)
 
+	os.Setenv("OCICRYPT_OAEP_HASHALG", "sha1")
```
Comment (lumjjb) [inline crypto/pkcs11/pkcs11helpers_test.go:136]: question: would it make sense to add another tests with sha256 as well?

File: crypto/pkcs11/pkcs11helpers.go [hunk 3]
```diff
@@ -333,7 +331,7 @@ func privateDecryptOAEP(privKeyObj *Pkcs11KeyFileObject, ciphertext []byte, hash
 
 	var oaep *pkcs11.OAEPParams
 
-	// the default is sha1
+	// An empty string from the Hash in the JSON defaults to sha1.
```
Comment (lumjjb) [inline crypto/pkcs11/pkcs11helpers.go:334]: nit: looks like the file originally was not `go fmt`'ed, indent is weird, but can fix in another PR. just noting it.

**Review by lumjjb — APPROVED** [review]
Minor non-blocking nits, lgtm

File: crypto/pkcs11/pkcs11helpers_test.go (hunk 2 above)
Comment (stefanberger) [reply crypto/pkcs11/pkcs11helpers_test.go:136]: a test with SoftHSM and sha256 doesn't work. See comments on related issue #60.

File: crypto/pkcs11/pkcs11helpers.go (hunk 3 above)
Comment (stefanberger) [reply crypto/pkcs11/pkcs11helpers.go:334]: I don't see `gofmt -w -s` reformatting anything on this file.

File: crypto/pkcs11/pkcs11helpers.go (hunk 1 above)
Comment (stefanberger) [reply crypto/pkcs11/pkcs11helpers.go:438]: Fixed

File: crypto/pkcs11/pkcs11helpers_test.go (hunk 2 above)
Comment (stefanberger) [reply crypto/pkcs11/pkcs11helpers_test.go:136]: https://github.com/containers/ocicrypt/issues/60#issuecomment-1111450382

File: crypto/pkcs11/pkcs11helpers.go (hunk 3 above)
Comment (lumjjb) [reply crypto/pkcs11/pkcs11helpers.go]: weird, maybe github formatting is doing weird things for me,

**Comment by lsm5** [comment]
Thanks a lot @stefanberger @lumjjb for this. Can we get new release with this please?

**Comment by arronwy** [comment]
> Before we release, want to make sure that we capture this change in the rust version as well @arronwy , any concerns of this from the rust side?

@lumjjb pkcs11 keywrapper is not implemented in rust version yet, this default setting will not impact current rust version.

**Pull request merged**